From e241ba7a51398626c880ec0045d34582f2b94d72 Mon Sep 17 00:00:00 2001
From: Y Vo <yvo@apm.com>
Date: Fri, 26 Jun 2015 15:01:47 +0700
Subject: [PATCH 04/41] arm64: dts: Add poweroff button device node for APM
 X-Gene platform
Git-commit: 3d8cc14152a4b661580761049c7b9711efa60d82
Patch-mainline: v4.2-rc3

This patch adds poweroff button device node to support poweroff feature
on APM X-Gene Mustang platform.

Signed-off-by: Y Vo <yvo@apm.com>
Signed-off-by: Kevin Hilman <khilman@linaro.org>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>

---
 arch/arm64/boot/dts/apm/apm-mustang.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/apm/apm-mustang.dts b/arch/arm64/boot/dts/apm/apm-mustang.dts
index 83578e7..4c55833 100644
--- a/arch/arm64/boot/dts/apm/apm-mustang.dts
+++ b/arch/arm64/boot/dts/apm/apm-mustang.dts
@@ -23,6 +23,16 @@
 		device_type = "memory";
 		reg = < 0x1 0x00000000 0x0 0x80000000 >; /* Updated by bootloader */
 	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		button@1 {
+			label = "POWER";
+			linux,code = <116>;
+			linux,input-type = <0x1>;
+			interrupts = <0x0 0x2d 0x1>;
+		};
+	};
 };
 
 &pcie0clk {
-- 
2.5.0.234.gefc8a62

